tests/testthat/test-serialization.R

```diff
@@ -241,7 +243,7 @@ test_that("collect() can retrieve NULL data types as NAs", {
 
   expect_equal(
     spark_types,
-    hive_type %>% pull(!! if(using_arrow()) "atype" else "rtype")
+    hive_type %>% pull(!! if(arrow_compat) "atype" else "rtype")
```

Comment:
I think we want to keep this one since it's use to choose between the arrow types and the r types in the expected tests data frame.

Instead, we probably want to add an IF statement like:

```
if (using_arrow()) {	
    # Disable while tracking fix for ARROW-3794	
    hive_type <- hive_type %>% filter(stype != "tinyint")	
  }
```

to override the new `arrowbigint` functionality in newer versions of R.